uisdk/src/main/java/com/karhoo/uisdk/screen/booking/booking/payment/braintree/BraintreePaymentPresenter.kt

```diff
@@ -127,30 +142,41 @@ class BraintreePaymentPresenter(view: PaymentMVP.View,
         }
     }
 
-    override fun onSavedPaymentInfoChanged(userPaymentInfo: SavedPaymentInfo?) {
-        view?.bindPaymentDetails(savedPaymentInfo = userPaymentInfo)
+    private fun passBackThreeDSecureNonce(nonce: String, amount: String) {
+        if (KarhooUISDKConfigurationProvider.handleBraintree()) {
+            view?.threeDSecureNonce(braintreeSDKToken)
+        } else {
+            view?.threeDSecureNonce(braintreeSDKToken, nonce, amount)
+        }
     }
 
-    override fun updateCardDetails(nonce: String, cardNumber: String?, cardTypeLabel: String?,
-                                   paymentData: String?) {
-        this.nonce = nonce
-        if (cardNumber != null && cardTypeLabel != null) {
-            userStore.savedPaymentInfo = SavedPaymentInfo(cardNumber, CardType.fromString
-            (cardTypeLabel))
-        }
-        view?.refresh()
+    private fun quotePriceToAmount(price: QuotePrice?): String {
+        val currency = Currency.getInstance(price?.currencyCode?.trim())
+        return CurrencyUtils.intToPriceNoSymbol(currency, price?.highPrice.orZero())
     }
 
-    override fun initialiseGuestPayment(price: QuotePrice?) {
-        view?.threeDSecureNonce(braintreeSDKToken, nonce, quotePriceToAmount(price))
+    override fun sdkInit(price: QuotePrice?) {
+        //currency is temporarily hardcoded to GBP as it isn't used by the backend to fix DROID-1536. Also hardcoded to GBP in the iOS code.
+        val sdkInitRequest = getSDKInitRequest("GBP")
```

Comment:
"GBP" is used in multiple places as the default currency, perhaps we can extract it and put it somewhere as a global constant?

Reply:
Done